src/main/java/algorithms/hard/MaximumSumof3NonOverlappingSubarrays.java

```diff
@@ -0,0 +1,57 @@
+package algorithms.hard;
+
+public class MaximumSumof3NonOverlappingSubarrays {
+
+    public int[] maxSumOfThreeSubarrays(int[] nums, int k) {
+        int n = nums.length;
+        int[] sum = new int[n + 1], leftBestIdx = new int[n], rightBestIdx = new int[n];
+
+        for (int i = 0; i < n; i++) {
+            sum[i + 1] = sum[i] + nums[i];
+        }
+
+        for (int i = k, maxSum = sum[k] - sum[0]; i < n; i++) {
+            if (sum[i + 1] - sum[i + 1 - k] > maxSum) {
+                leftBestIdx[i] = i + 1 - k;
+                maxSum = sum[i + 1] - sum[i + 1 - k];
+            } else {
+                leftBestIdx[i] = leftBestIdx[i - 1];
+            }
+        }
+
+        rightBestIdx[n - k] = n - k;
+        for (int i = n - k - 1, maxsum = sum[n] - sum[n - k]; i >= 0; i--) {
+            if (sum[i + k] - sum[i] >= maxsum) {
```

Comment:
Greater or equal because of the lexographical order again. We need to choose left-most one if the sums are same.